app/models/qc_results_upload_factory.rb

```diff
@@ -144,13 +149,16 @@ def qc_decision
   def pivot_csv_data_to_obj
     header_converter = proc do |header|
       assay_type = QcAssayType.find_by(label: header.strip)
-      assay_type ? assay_type.key : header
+      assay_type ? assay_type.key : header.strip
     end
 
-    csv = CSV.new(csv_string_without_groups, headers: true, header_converters: header_converter,
-                                             converters: :all)
+    csv = CSV.new(csv_string_without_groups, headers: true, header_converters: header_converter)
 
-    csv.to_a.map(&:to_hash)
+    arr_of_hashes = csv.to_a.map(&:to_hash)
```

Comment:
I think a bit of documentation would be good just to explain why we are excluding these lines?